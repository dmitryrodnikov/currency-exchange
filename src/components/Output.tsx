import * as React from 'react';
import styled from 'styled-components';
import {FONT_SIZE} from '../constants/Theme';

const StyledOutput = styled.div`
    font-size: ${FONT_SIZE.l};
    font-weight: bold;
    text-align: right;
    padding: 0 8px;
`;

export const Output: React.FunctionComponent<OutputProps> = ({text}) => {
    return (
        <StyledOutput>{text}</StyledOutput>
    );
};

interface OutputProps {
    text: string;
}
