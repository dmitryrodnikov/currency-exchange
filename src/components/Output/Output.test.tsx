import React from 'react';
import {shallow, render} from 'enzyme';
import {Output} from './Output';

const VALUE= 'Test';

describe('Output', () => {
    it('Shallow render snapshots are equal', () => {
        const wrapper = shallow(
            <Output text={VALUE}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('Render snapshots are equal', () => {
        const wrapper = render(
            <Output text={VALUE}/>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
