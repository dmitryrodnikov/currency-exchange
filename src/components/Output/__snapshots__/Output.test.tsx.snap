// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output Render snapshots are equal 1`] = `
<div
  class="sc-bdVaJa iQlKOp"
>
  Test
</div>
`;

exports[`Output Shallow render snapshots are equal 1`] = `
<styled.div>
  Test
</styled.div>
`;
